code/tests/utilities/test_AzureBlobStorageClient.py

```diff
@@ -46,6 +46,51 @@ def test_file_exists(BlobServiceClientMock: MagicMock, exists: bool, expected: b
     )
 
 
+@patch("backend.batch.utilities.helpers.AzureBlobStorageClient.generate_blob_sas")
+@patch("backend.batch.utilities.helpers.AzureBlobStorageClient.BlobServiceClient")
+@pytest.mark.parametrize("content_type", [(None), ("text/pdf")])
+def test_upload_file(
+    BlobServiceClientMock: MagicMock,
+    generate_blob_sas_mock: MagicMock,
+    content_type: str,
+):
+    # given
+    client = AzureBlobStorageClient()
+    blob_service_client_mock = BlobServiceClientMock.from_connection_string.return_value
+    blob_client_mock = blob_service_client_mock.get_blob_client.return_value
+    blob_client_mock.url = "mock_url"
+    generate_blob_sas_mock.return_value = "mock-sas"
+
+    # when
+    result = client.upload_file(str.encode("mock-data"), "mock-file", content_type)
+
+    # then
+    blob_client_mock.upload_blob.assert_called_once_with(
+        str.encode("mock-data"),
+        overwrite=True,
+        content_settings=ANY,
+        metadata={"title": "mock-file"},
+    )
+    _, kwargs = blob_client_mock.upload_blob.call_args
+    assert (
+        kwargs["content_settings"]["content_type"] == content_type
+        if content_type is not None
```

Comment:
I try to avoid if conditions in tests where possible and keep any logic to a minimum. It tends to make them harder to follow and maintain.

Could you use a tuple on line 51 instead to also pass in the expectation? You can see an example here https://docs.pytest.org/en/7.1.x/how-to/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions